app/models/manageiq/providers/hawkular/middleware_manager/refresh_parser.rb

```diff
@@ -187,6 +187,17 @@ def process_availability(availability)
         end
       end
 
+      def parse_availability(availabilities, resources_by_metric_id)
+        availabilities.map! do |availability|
+          resources_by_metric_id[availability['id']][:status] = process_availability(availability['data'].first)
+          availability['id']
+        end
+        (resources_by_metric_id.keys - availabilities).each do |metric_id|
```

Comment:
Availabilities requested minus received.

I'm iterating over the ids that aren't on the availability (the ones that didn't have any metric) to set the status to Unknown.

I agree is not very clear. Maybe I could add more variables to make it clear.